### ISSUE TITLE
Parallelize tasks that can be run in parallel

### ISSUE THREAD

**Issue by mkavulich** [NCAR/wrfcloud#78]
## Describe the Enhancement ##
 - Pulling GFS data: each forecast hour can be pulled in parallel
 - ungrib.exe: each file can be "ungribbed" in parallel
 - metgrid.exe: each metgrid file can be processed as soon as ungrib.exe completes for that hour
 - UPP/graphics: each forecast hour can be processed as soon as the WRF output is written to disk

### Time Estimate ###
1 day per sub-issue

### Sub-Issues ###
Consider breaking the enhancement down into sub-issues.
- [ ] Pull GFS data for each forecast hour in parallel
- [ ] ungrib each GFS input file in parallel
- [ ] process metgrid files once ungrib completes for a given hour

### Relevant Deadlines ###
NONE

### Assignee ###
- [ ] Select **engineer(s)** or **no engineer** required
- [ ] Select **scientist(s)** or **no scientist** required

### Labels ###
- [ ] Select **component(s)**
- [ ] Select **priority**

### Projects and Milestone ###
- [ ] Select **Project**
- [ ] Select **Milestone** as the next official version or **Backlog of Development Ideas**

## Enhancement Checklist ##
- [ ] Complete the issue definition above, including the **Time Estimate** and **Funding Source**.
- [ ] Fork this repository or create a branch of **develop**.
Branch name: `feature_<Issue Number>/<Description>`
- [ ] Complete the development and test your changes.
- [ ] Add/update log messages for easier debugging.
- [ ] Add/update tests.
- [ ] Add/update documentation.
- [ ] Push local changes to GitHub.
- [ ] Submit a pull request to merge into **develop**.
Pull request: `feature <Issue Number> <Description>`
- [ ] Define the pull request metadata, as permissions allow.
Select: **Reviewer(s)**, **Project**, and **Development** issue
Select: **Milestone** as the next official version
- [ ] Iterate until the reviewer(s) accept and merge your changes.
- [ ] Delete your fork or branch.
- [ ] Close this issue.
